src/modules/oAuth/oAuth.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { UsersRepository } from '../users/users.repository';
+import { JwtService } from '@nestjs/jwt';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class OAuthService {
+  constructor(
+    private userRepository: UsersRepository,
+    private readonly jwtService: JwtService,
+    private readonly configService: ConfigService,
+  ) {}
+  async googleLogin(payload: any) {
+    let user = await this.userRepository.getUserBy({
+      email: payload?.email,
+    });
+
+    if (!user) {
+      user = await this.userRepository.createUser({
+        email: payload?.email,
+        firstName: payload?.firstName,
+        lastName: payload?.lastName,
+        password: '',
```

Comment:
is password field nullable in entity ?

Reply:
it is not neither nullable nor required, and I did not know how to create password, that is why i just sent empty string